arbos/parse_l2.go

```diff
@@ -171,6 +184,21 @@ func parseL2Message(rd io.Reader, poster common.Address, timestamp uint64, reque
 			return nil, types.ErrTxTypeNotSupported
 		}
 		return types.Transactions{newTx}, nil
+	case L2MessageKind_EspressoTx:
+		newTx := new(types.Transaction)
+		// Safe to read in its entirety, as all input readers are limited
+		readBytes, err := io.ReadAll(rd)
+		if err != nil {
+			return nil, err
+		}
+		if err := json.Unmarshal(readBytes, &newTx); err != nil {
```

Comment:
It seems the only difference between this and the regular `L2MessageKind_SignedTx` above is that the Espresso txs are encoded as JSON and the regular transactions are encoded as binary. Why not just use the binary encoding for Espresso transactions and then we don't need separate parsing for them?

Reply:
Ah yes you're right, I agree, we should only need a special message type for the batch as a whole

Reply:
Yes, you are right. I didn't realize that we should do the parsing in a totally different way.
I am not sure what is your question about JSON fomrat. We make a forwarding when receving the user transactions [here](https://github.com/EspressoSystems/nitro-espresso-integration/blob/1a0a306bb6219eacfdc7c1e18f15837c2401f5c7/arbos/espresso/client.go#L64). And user txs is encoded in JSON before sending to espresso sequencer.

Reply:
I see. I don't think we should be using JSON there either. JSON is fine for encoding the _Espresso_ transaction, but when we encode the _Arbitrum_ transaction [here](https://github.com/EspressoSystems/nitro-espresso-integration/blob/1a0a306bb6219eacfdc7c1e18f15837c2401f5c7/arbos/espresso/client.go#L65), I think we should use `MarshalBinary`, which should use the standard RLP encoding that you would get if, e.g. you submitted a transaction via the usual web3 APIs. We gain nothing and waste a lot of data by keeping EVM transactions JSON-encoded throughout the entire stack.

However, since we are currently being consistent in using JSON everywhere, it seems that this doesn't need to block this PR. We can merge it as-is and switch to a binary encoding later. @ImJeremyHe can you create an issue to follow up on this?